src/test/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.idam.api.external.model.ActionMap;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesRequest;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesResponse;
+import uk.gov.hmcts.reform.idam.api.external.model.EvaluatePoliciesResponseInner;
+import uk.gov.hmcts.reform.idam.api.external.model.Subject;
+import uk.gov.hmcts.reform.idam.web.config.properties.ConfigurationProperties;
+
+import java.util.Collections;
+
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.springframework.http.ResponseEntity.ok;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PolicyServiceTest {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    ConfigurationProperties configurationProperties;
+    @Mock
+    RestTemplate restTemplate;
+    @InjectMocks
+    PolicyService service;
+
+    @Before
+    public void setupProps() {
+        given(configurationProperties.getStrategic().getPolicies().getApplicationName())
+            .willReturn("applicationName");
+        given(configurationProperties.getStrategic().getService().getUrl())
+            .willReturn("idamApi");
+        given(configurationProperties.getStrategic().getEndpoint().getEvaluatePolicies())
+            .willReturn("evaluatePolicies");
+    }
+
+    EvaluatePoliciesResponse mockResponse(final ActionMap actionMap) {
+        final EvaluatePoliciesResponse response = new EvaluatePoliciesResponse();
+        response.add(new EvaluatePoliciesResponseInner()
+            .actions(actionMap));
+        return response;
+    }
+
+    EvaluatePoliciesRequest expectedRequest(String resource, String application, String ssoToken, String ipAddress) {
+        return new EvaluatePoliciesRequest()
+            .resources(singletonList("someUri"))
+            .application("applicationName")
+            .subject(new Subject().ssoToken("someToken"))
+            .environment(ImmutableMap.of("requestIp", singletonList("someIpAddress")));
+    }
+
+    HttpHeaders expectedHeaders(String token) {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.put(HttpHeaders.COOKIE, Collections.singletonList("Idam.Session=" + token));
```

Comment:
Is there not a constant for Idam.Session somewhere? be annoying if we ever need to rename it again